feat(router): support for object in queryParams

soc221b opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #47307


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

JeanMeche commented:
Hi, before we have a look to this change can you make sure to sign the CLA, thank you.

soc221b commented:
Hi, I've signed the CLA 1 day ago, but it is happened after opening this PR. Do I need to make my email public or do we just need to trigger the cla/google workflow again?

Thanks.

AndrewKushnir commented:
@soc221b I've triggered a rerun on the CLA check and it's still failing, [see here](https://github.com/angular/angular/pull/58580/checks?check_run_id=32896420260). Could you please confirm that the email that you've used there is the same one as your GitHub account?

soc221b committed:
feat(router): support for object in queryParams

related issue: #47307

soc221b committed:
refactor(router): the tree doesn't stringify quary params now

the tree will fully clone query params, we don't need to assign to the
original

soc221b committed:
test: rollback to original tests

atscott commented:
[TGP](https://fusion2.corp.google.com/presubmit/720242782/OCL:720242782:BASE:720180594:1738024322903:600ea433/targets) shows 21 failures, many of which are due to 0-1 query. There are a handful of others that will need some more digging. 

The results here sound quite similar to the [initial attempt](https://github.com/angular/angular/issues/47307#issuecomment-1258225444) which ended up have unsolvable blockers. This also doesn't address matrix parameters, [which have the same problem](https://github.com/angular/angular/issues/47307#issuecomment-1734277747)

atscott commented:
Closing, per https://github.com/angular/angular/pull/58580#issuecomment-2622389064
